Automattic/wp-calypso
# Restrict the domain selection for sites created via /setup/link-in-bio-tld

#### Proposed Changes

_This PR is based on #71531 , and is blocked by https://github.com/Automattic/jetpack/pull/28087 and code-D96551._

This PR implements the mechanism for restricting the available domain options to specific TLD in `/domains/add`. At the moment, it's specifically for the sites created via the dotlink partnership version of Link in Bio flow, i.e. `/setup/link-in-bio-tld`, while the code structure itself should be generic enough for future expansion. More details can be found in Automattic/martech#1320.

A similar case in the site launch flow will be addressed in a follow-up PR.

#### Testing Instructions

1. Go through /setup/link-in-bio-tld/ to create a Link in Bio site.
2. In the Launchpad, click "customize URL" to access the `/domains/add` page.
3. There should only be `.link` and `w.link` domains available.



*

#### Pre-merge Checklist
